src/main/java/de/nplay/moderationbot/moderation/ModerationService.java

```diff
@@ -0,0 +1,148 @@
+package de.nplay.moderationbot.moderation;
+
+import de.chojo.sadu.mapper.annotation.MappingProvider;
+import de.chojo.sadu.mapper.rowmapper.RowMapping;
+import de.chojo.sadu.queries.api.call.Call;
+import de.chojo.sadu.queries.api.query.Query;
+import de.chojo.sadu.queries.api.results.writing.insertion.InsertionResult;
+import de.nplay.moderationbot.rules.RuleService;
+
+import javax.annotation.Nullable;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.Optional;
+
+/**
+ * Utility class for managing moderations.
+ */
+public class ModerationService {
+
+    /**
+     * Record representing a moderation.
+     *
+     * @param id               The unique identifier of the moderation.
+     * @param userId           The ID of the user being moderated.
+     * @param type             The type of moderation.
+     * @param reverted         Whether the moderation has been reverted.
+     * @param reason           The reason for the moderation.
+     * @param paragraph        The rule paragraph associated with the moderation.
+     * @param referenceMessage The reference message associated with the moderation.
+     * @param revoke_at        The timestamp when the moderation will be revoked.
+     * @param duration         The duration of the moderation.
+     * @param issuerId         The ID of the user who issued the moderation.
+     * @param created_at       The date when the moderation was created.
+     */
+    public record Moderation(
```

Comment:
This name is ambiguous, maybe we should rename the table to `reports` and thus the mapping record to `Report`

Reply:
IMO Reports would be even more confusing, since reports are generally things that are triggered by "normal" guild members to inform the moderators, that something against the rules is happening

Reply:
In feel `ModerationAct` could be a good term, but let me sleep on it for a night